### PR TITLE
[Feature] Add USD price impact

### PR DESCRIPTION
**Pull request by berteotti**
- Creates warningFiatSeverity
- Creates computeFiatValuePriceImpact
- Display fiat price impact beside fiat price
- Remove max button
- Display balance below token input
Screenshot 2022-02-21 at 10 19 58

### CODE REVIEW

File: src/constants/index.tsx [hunk 1]
```diff
@@ -252,6 +252,8 @@ export const ALLOWED_PRICE_IMPACT_HIGH: Percent = new Percent(JSBI.BigInt(500),
 export const PRICE_IMPACT_WITHOUT_FEE_CONFIRM_MIN: Percent = new Percent(JSBI.BigInt(1000), BIPS_BASE) // 10%
 // for non expert mode disable swaps above this
 export const BLOCKED_PRICE_IMPACT_NON_EXPERT: Percent = new Percent(JSBI.BigInt(1500), BIPS_BASE) // 15%
+// used for fiat warning states
+export const ALLOWED_FIAT_PRICE_IMPACT_HIGH: Percent = new Percent(JSBI.BigInt(200), BIPS_BASE) // 2%
```
Comment (adamazad) [inline src/constants/index.tsx:256]: I would say 2% is high. We will get back to this when @zamli is back.

File: src/utils/prices.ts [hunk 2]
```diff
@@ -97,6 +98,11 @@ export function warningSeverity(priceImpact: Percent | undefined): 0 | 1 | 2 | 3
   return 0
 }
 
+export function warningFiatSeverity(priceImpact: Percent | undefined): 0 | 1 {
+  if (!priceImpact?.lessThan(ALLOWED_FIAT_PRICE_IMPACT_HIGH)) return 1
```
Comment (adamazad) [inline src/utils/prices.ts:102]: We should thrive for consistency. `warningSeverity` returns `3` when price impact is high. 

`ALLOWED_FIAT_PRICE_IMPACT_HIGH` should map to 3 or be changed to `_LOW`

File: src/pages/Swap/index.tsx [hunk 3]
```diff
@@ -301,9 +305,8 @@ export default function Swap() {
           <AutoColumn gap="12px">
             <AutoColumn gap="3px">
               <CurrencyInputPanel
-                label={independentField === Field.OUTPUT && !showWrap && trade ? 'From (estimated)' : 'From'}
                 value={formattedAmounts[Field.INPUT]}
-                showMaxButton={!atMaxAmountInput}
+                showMaxButton={false}
```
Comment (adamazad) [inline src/pages/Swap/index.tsx]: We probably won't need this property.

File: src/components/FiatValueDetails/index.tsx [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+  fiatValue?: CurrencyAmount | null
+  priceImpact?: Percent
+}
+
+const PriceImpactText = styled.span<{ warning?: boolean }>`
+  color: ${({ theme, warning }) => warning && theme.red1};
+  margin-left: 8px;
+`
+
+export default function FiatValueDetails({ fiatValue, priceImpact }: FiatValueDetailsProps) {
```
Comment (adamazad) [inline src/components/FiatValueDetails/index.tsx:17]:
```suggestion
export function FiatValueDetails({ fiatValue, priceImpact }: FiatValueDetailsProps) {
```

File: src/components/FiatValueDetails/index.tsx [hunk 5]
```diff
@@ -0,0 +1,30 @@
...
+import { TYPE } from '../../theme'
+import { CurrencyAmount, JSBI, Percent } from '@swapr/sdk'
+import { warningFiatSeverity } from '../../utils/prices'
+
+interface FiatValueDetailsProps {
+  fiatValue?: CurrencyAmount | null
+  priceImpact?: Percent
+}
+
+const PriceImpactText = styled.span<{ warning?: boolean }>`
```
Comment (adamazad) [inline src/components/FiatValueDetails/index.tsx:12]:
```suggestion
const StyledPriceImpact = styled.span<{ warning?: boolean }>`
```

File: src/components/FiatValueDetails/index.tsx [hunk 6]
```diff
@@ -0,0 +1,30 @@
+import React from 'react'
+import styled from 'styled-components'
+import { TYPE } from '../../theme'
+import { CurrencyAmount, JSBI, Percent } from '@swapr/sdk'
+import { warningFiatSeverity } from '../../utils/prices'
+
+interface FiatValueDetailsProps {
+  fiatValue?: CurrencyAmount | null
```
Comment (adamazad) [inline src/components/FiatValueDetails/index.tsx:8]: Should be required or default to `0` in component properties

**Review by adamazad — CHANGES_REQUESTED** [review]

File: src/components/FiatValueDetails/index.tsx (hunk 6 above)
Comment (berteotti) [reply src/components/FiatValueDetails/index.tsx:8]: Should `fiatValue` be computed to a string in the parent component, so we can have `0` as default in component properties? In the other scenario I think having 3 different valid types is a little bit messy, like `CurrencyAmount | string | null`

File: src/constants/index.tsx (hunk 1 above)
Comment (berteotti) [reply src/constants/index.tsx:256]: I don't know if this helps but in the issue is defined 2% for "high price impact". https://github.com/levelkdev/dxswap-dapp/issues/640

File: src/utils/prices.ts (hunk 2 above)
Comment (berteotti) [reply src/utils/prices.ts:102]: Think that we should keep the name `_HIGH` as the intention is to display high risk. Mapping to `3` in this one.

File: src/components/CurrencyInputPanel/index.tsx [hunk 7]
```diff
@@ -12,6 +12,7 @@ import { ReactComponent as DropDown } from '../../assets/images/dropdown.svg'
 
 import { useActiveWeb3React } from '../../hooks'
 import { useTranslation } from 'react-i18next'
+import FiatValueDetails from '../FiatValueDetails'
```
[the remark this answers is not in the document]
Comment (berteotti) [reply src/components/CurrencyInputPanel/index.tsx:15]: I'm curious about why we should prefer named exports 🙂

**Pull request merged**